packages/DataGrid/src/DataGrid.js

```diff
@@ -500,12 +497,12 @@ const DataGrid = React.forwardRef((props, ref) => {
         onFocus={handleFocusGrid}
         onKeyDown={handleKeyDownGrid}
         onKeyUp={handleKeyUpGrid}
-        onMouseUp={handleMouseUpGrid}
+        onMouseDown={handleMouseDownGrid}
         onMouseOver={handleMouseOver}
         onMouseLeave={handleMouseLeave}
         ref={refContainer}
         role="grid"
-        tabIndex={gridShouldHaveFocus ? 0 : -1}
+        tabIndex={0}
```

Comment:
I remove gridShouldHaveFocus because it was preventing from clickling the first time the Datagrid, also I think there should be a better discussion about how to handle the `tab` process if the Datagrid it's part of a form. `a11y` discussion. cc @mikrotron 

We can improve and bring back this feature later.